tests/flask/test_oauth1/test_resource_protector.py

```diff
@@ -1,170 +1,178 @@
-        auth_header = auth_header.replace("valid-token-secret", "invalid")
-        headers = {"Authorization": auth_header}
-        rv = self.client.get(url, headers=headers)
-        data = json.loads(rv.data)
-        assert data["error"] == "invalid_signature"
-
-    def test_hmac_sha1_signature(self):
-        self.prepare_data()
-        url = "/user"
-
-        params = [
-            ("oauth_consumer_key", "client"),
-            ("oauth_token", "valid-token"),
-            ("oauth_signature_method", "HMAC-SHA1"),
-            ("oauth_timestamp", str(int(time.time()))),
-            ("oauth_nonce", "hmac-sha1-nonce"),
-        ]
-        base_string = signature.construct_base_string(
-            "GET", "http://localhost/user", params
-        )
-        sig = signature.hmac_sha1_signature(base_string, "secret", "valid-token-secret")
-        params.append(("oauth_signature", sig))
-        auth_param = ",".join([f'{k}="{v}"' for k, v in params])
-        auth_header = "OAuth " + auth_param
-        headers = {"Authorization": auth_header}
-
-        # case 1: success
-        rv = self.client.get(url, headers=headers)
-        data = json.loads(rv.data)
-        assert "username" in data
-
-        # case 2: exists nonce
-        rv = self.client.get(url, headers=headers)
-        data = json.loads(rv.data)
-        assert data["error"] == "invalid_nonce"
-
-    def test_rsa_sha1_signature(self):
-        self.prepare_data()
-        url = "/user"
-
-        params = [
-            ("oauth_consumer_key", "client"),
-            ("oauth_token", "valid-token"),
-            ("oauth_signature_method", "RSA-SHA1"),
-            ("oauth_timestamp", str(int(time.time()))),
-            ("oauth_nonce", "rsa-sha1-nonce"),
-        ]
-        base_string = signature.construct_base_string(
-            "GET", "http://localhost/user", params
-        )
-        sig = signature.rsa_sha1_signature(
-            base_string, read_file_path("rsa_private.pem")
+@pytest.fixture(autouse=True)
+def user(db):
+    user = User(username="foo")
+    db.session.add(user)
+    db.session.commit()
+    yield user
+    db.session.delete(user)
+
+
+@pytest.fixture(autouse=True)
+def client(db, user):
+    client = Client(
+        user_id=user.id,
+        client_id="client",
+        client_secret="secret",
+        default_redirect_uri="https://a.b",
+    )
+    db.session.add(client)
+    db.session.commit()
+    yield client
+    db.session.delete(client)
+
+
+@pytest.fixture(autouse=True)
+def token(db, user, client):
+    tok = TokenCredential(
+        user_id=user.id,
+        client_id=client.client_id,
+        oauth_token="valid-token",
+        oauth_token_secret="valid-token-secret",
+    )
+    db.session.add(tok)
+    db.session.commit()
+
+
+@pytest.mark.parametrize("use_cache", [True, False])
+def test_invalid_request_parameters(app, test_client, use_cache):
+    create_resource_server(app, use_cache, use_cache)
+    url = "/user"
+
+    # case 1
```

Comment:
Suggestion: Instead of having `case 1... case 5` all in one test, use `pytest.mark.parametrize()` to separate these into individual test cases.

Reply:
I like pytest parametrize, but in that particular case pytest it a bit overkill to me. There are 3 vars to parametrize (url, error, error_description) for a 4 lines test. Anyway, I think both solutions would be fine.
I did not put a lot of factorization effort into OAuth1, but feel free to open a PR if you want to work on this.